src/operators/querytools.h

```diff
@@ -9,8 +10,11 @@
  */
 class QueryTools {
 	public:
-		explicit QueryTools(QueryProfiler &profiler) : profiler(profiler) {};
+        explicit QueryTools(QueryProfiler &profiler) : profiler(profiler) {};
+		QueryTools(QueryProfiler &profiler, std::shared_ptr<UserDB::Session> session) : profiler(profiler), session(session) {};
```

Comment:
Wenn du den Per-Value reingibst wie hier, dann ist der ja schon kopiert.
```suggestion
		QueryTools(QueryProfiler &profiler, std::shared_ptr<UserDB::Session> session) : profiler(profiler), session(std::move(session)) {};
```

Reply:
das wird bestimmt wegoptimiert